packages/benchmarks/lit-html/kitchen-sink/index-compiled.ts

```diff
@@ -0,0 +1,246 @@
+// import { repeat } from "lit-html/lib/directives/repeat.js";
+const repeat = (items: any[], _kfn: (i: any) => any, tfn: (i: any) => any) =>
+  items.map((i) => tfn(i));
+
+// TODO(kschaaf) use real classMap once landed
+// import { classMap } from "lit-html/lib/directives/class-map.js";
+const classMap = (classes: {[key: string]: boolean}) =>
+  Object.keys(classes)
+    .filter((k) => classes[k])
+    .join(' ');
+
+// TODO(kschaaf) use real styleMap once landed
+// import { styleMap } from "lit-html/lib/directives/style-map.js";
+const styleMap = (style: {[key: string]: boolean}) =>
+  Object.entries(style)
+    .map(([k, v]) => `${k}: ${v}`)
+    .join('; ');
+
+// IE doesn't support URLSearchParams
+const params = document.location.search
+  .slice(1)
+  .split('&')
+  .map((p) => p.split('='))
+  .reduce(
+    (p: {[key: string]: any}, [k, v]) => ((p[k] = JSON.parse(v || 'true')), p),
+    {}
+  );
+
+// Configurable params
+const width = params.width ?? 4;
+const depth = params.depth ?? 4;
+const updateCount = params.updateCount ?? 10;
+const nopUpdateCount = params.nopUpdateCount ?? 10;
+
+// Data model
+interface Data {
+  key: number;
+  text: string;
+  property: {};
+  classes: {};
+  style: {};
+  node: Node;
+  handler: () => void;
+  childData?: Data[];
+  useRepeat?: boolean;
+}
+
+/**
+ * Generates a data model for an item, recursively creating child data
+ * models for the given width & depth.
+ *
+ * @param updateId Increment to ensure new data models for the given id
+ *   create unique (non-dirty checking) values
+ * @param id Id for item, unique amongst its peers
+ * @param parent Parent moniker (to create unique text for each item)
+ * @param currentDepth Current depth, used to stop recursion at REPEAT_DEPTH.
+ */
+const generateData = (
+  updateId = 0,
+  id = 0,
+  parent: string | undefined = undefined,
+  currentDepth = 0
+): Data => {
+  const moniker = `${parent ? `${parent}-` : ''}${id}`;
+  return {
+    key: id,
+    text: `Item ${moniker}${updateId ? `#${updateId}` : ''}`,
+    property: {},
+    node: document.createElement('span'),
+    classes: {
+      foo: !!(updateId % 2),
+      bar: !(updateId % 2),
+      ziz: true,
+      zaz: false,
+    },
+    style: {
+      'border-top-width': `${updateId % 2}px`,
+      'border-bottom-width': `${(updateId + 1) % 2}px`,
+      'border-left-width': '0px',
+      'border-right-width': '0px',
+    },
+    handler: () => {},
+    ...(currentDepth < depth && {
+      childData: new Array(width)
+        .fill(0)
+        .map((_, i) => generateData(updateId, i, moniker, currentDepth + 1)),
+      useRepeat: !(id % 2),
+    }),
+  };
+};
+
+// These are "hand compiled" templates for the benchmark. This allows lit-html
+// to skip the template parsing/prep steps for faster bootup.
+const _$lit_template_1$ = {
+  _element: document.createElement('template'),
+  _parts: [
+    {_type:1,_index:1,_name:"class",_strings:[""," static"], _constructor: AttributePart},
```

Comment:
Since we already have the `PartType` enum, we could use those numbers here to indirect the constructor, but then we'd need to add code in the core to have that switch.

I guess this could also compile to `get _constructor: window.litGetPartCtor(1)` where that is installed somehow.

Reply:
That brings up a separate point: all of these names will be mangled right? Seems like we need a gambit for that. Make them all `stableProperties` and the compiler compiles to that?

Reply:
Stable properties sounds right, since the compiled templates should work with dev and prod

Reply:
Do you see anything particularly bad with using private-ssr-support for this? We could rename that...

Reply:
I added relevant properties to stable properties list. Still using private-ssr-support to provide the stable name for TemplateImpl and part constructors. We could make stable properties for them too though, with little code-size impact.

Reply:
We might want to export Template and parts like https://github.com/Polymer/lit-html/pull/1692 did perviously, to avoid bringing in ssr support into compiled modules.